### PR TITLE
bpo-36268: Change default tar format to pax from GNU 

### LINKED ISSUE #80449: Change default tar format to modern POSIX 2001 (pax) for better portability/interop, support and standards conformance
BPO | [36268](https://bugs.python.org/issue36268)
--- | :---
Nosy | @gustaebel, @serhiy-storchaka, @CAM-Gerlach
PRs | <li>python/cpython#12355</li><li>python/cpython#12635</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2019-03-21.14:46:40.428>
created_at = <Date 2019-03-12.02:14:28.198>
labels = ['3.8', 'type-bug', 'library']
title = 'Change default tar format to modern POSIX 2001 (pax) for better portability/interop, support and standards conformance'
updated_at = <Date 2019-03-30.17:36:23.905>
user = 'https://github.com/CAM-Gerlach'
```

bugs.python.org fields:
```python
activity = <Date 2019-03-30.17:36:23.905>
actor = 'CAM-Gerlach'
assignee = 'none'
closed = True
closed_date = <Date 2019-03-21.14:46:40.428>
closer = 'serhiy.storchaka'
components = ['Library (Lib)']
creation = <Date 2019-03-12.02:14:28.198>
creator = 'CAM-Gerlach'
dependencies = []
files = []
hgrepos = []
issue_num = 36268
keywords = ['patch']
message_count = 9.0
messages = ['337710', '337860', '337871', '337929', '337951', '338020', '338151', '338546', '338547']
nosy_count = 3.0
nosy_names = ['lars.gustaebel', 'serhiy.storchaka', 'CAM-Gerlach']
pr_nums = ['12355', '12635']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue36268'
versions = ['Python 3.8']
```

</p></details>

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Per my proposal and the resulting discussion in [bpo-36268](https://bugs.python.org/issue36268), changes tarfile.DEFAULT_FORMAT to be tarfile.PAX_FORMAT , rather than the legacy tarfile.GNU_FORMAT for Python 3.8. It also updates the docs to reflect this, explicitly passes ``tarfile.GNU_FORMAT`` in one test that assumed that ``tarfile.DEFAULT_FORMAT`` would be this, and adds an appropriate news entry (I assume this is far too trivial to be included in What's New or myself in ``Misc/ACKS``).

This should also fix [bpo-30661](https://bugs.python.org/issue30661), and offers several other benefits, including fewer limitations, greater interoperability with POSIX-conformant implementations, increased extensibility and backward-compatibility with ustar, cross-platform portability with a standardized encoding (UTF-8) and avoidance of errors [like this one](https://stackoverflow.com/questions/19902544/tarfile-produce-garbled-file-name-in-the-tar-gz-archivement).

Fix #80449

https://bugs.python.org/issue36268

### CODE REVIEW

File: Doc/library/tarfile.rst
Comment (serhiy-storchaka) [inline Doc/library/tarfile.rst]: I think this needs additional indentation. "versionchanged" is corresponded to "DEFAULT_FORMAT", not to the whole module.

File: Misc/NEWS.d/next/Library/2019-03-14-16-25-17.bpo-36268.MDXLw6.rst
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2019-03-14-16-25-17.bpo-36268.MDXLw6.rst]: Add "Patch by *yourname*."

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Please add also a What's New entry.

File: Doc/library/tarfile.rst
Comment (CAM-Gerlach) [reply Doc/library/tarfile.rst]: Great catch, thanks. Done.

File: Misc/NEWS.d/next/Library/2019-03-14-16-25-17.bpo-36268.MDXLw6.rst
Comment (CAM-Gerlach) [reply Misc/NEWS.d/next/Library/2019-03-14-16-25-17.bpo-36268.MDXLw6.rst]: Okay, done.

**Comment by CAM-Gerlach** [comment]
@serhiy-storchaka I have made the requested changes; please review again.

Also, should I keep the NEWS entry commit separate (as I've seen others do in the most recent PRs here), or is that not necessary? Should I amend the commit message to include the now-known GH-NNNN number, or is that not needed for a non-backport? Thanks.

Changes:
* Added What's New entry
* Added name to NEWS entry
* Moved ``versionchanged`` to the correct indent level
* Minor textual refinements
* Re-added bpo-number to commit message (it evidently got removed in a rebase before opening the PR)

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. But before merging this we should ask the opinion of the module maintainer, @gustaebel.

**Review by gustaebel — APPROVED** [review]
Switching to PAX_FORMAT is a very good idea.

**Comment by CAM-Gerlach** [comment]
@serhiy-storchaka Perhaps @ncoghlan should also chime in, since he opened [bpo-30661](https://bugs.python.org/issue30661) which brought this change up as a possible solution?

**Comment by ncoghlan** [comment]
+1 from me (if I recall correctly, the PAX_FORMAT vs GNU_FORMAT question came up in a distutils-sig discussion about the sdist tar.gz format, and the fact that those should really be in `PAX_FORMAT`, but `shutil.make_archive` didn't support that)

**Comment by CAM-Gerlach** [comment]
Thanks @ncoghlan @serhiy-storchaka @gustaebel !

**Pull request merged**
